appliance_catalog/views.py

```diff
@@ -141,8 +143,24 @@ def get_appliance_template(request, pk):
     logger.info('Getting and displaying YAML template for appliance')
 
     appliance = Appliance.objects.filter(pk=pk).get()
-    return HttpResponse(appliance.template, content_type="text/yaml")
 
+    # send message to GA
+    if settings.GOOGLE_ANALYTICS_PROPERTY_ID:
+        if request.user and request.user.username:
```

Comment:
Do we have to attach the user's username to this? I'm not sure it's necessary, we can probably just use the random ID and not have to worry about storing things tied to user PII (personally identifiable information).

Reply:
For users, I thought GA would have some interesting statistics, such as downloads by users. But as far as I know, there aren't any statistics on events related to users. I can just use a random id as client id.